src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c

```diff
@@ -60,66 +60,206 @@ const ActorInit Door_Shutter_InitVars = {
-    /* 0x0008 */ u8 c;
-    /* 0x0009 */ u8 translateZ;
-    /* 0x000A */ u8 e;
-    /* 0x000B */ u8 f;
-} ShutterInfo;
-
-static ShutterInfo sShutterInfo[] = {
-    { gDTDungeonDoor1DL, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { gDTDungeonDoor2DL, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { gDodongoDoorDL, gDodongoBarsDL, 240, 14, 70, 15 },
-    { gJabuDoorSection1DL, gJabuWebDoorDL, 0, 110, 50, 15 },
-    { gPhantomGanonBarsDL, NULL, 130, 12, 50, 15 },
-    { gGohmaDoorDL, NULL, 130, 12, 50, 15 },
-    { gSpiritDoorDL, gJyaDoorMetalBarsDL, 240, 14, 50, 15 },
-    { gBossDoorDL, NULL, 130, 12, 50, 15 },
-    { gDungeonDoorDL, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { gFireTempleDoorFrontDL, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { gFireTempleDoorBackDL, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { object_ganon_objects_DL_0000C0, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { gObjectMizuObjectsDoorShutterDL_005D90, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { gObjectMizuObjectsDoorShutterDL_007000, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { object_haka_door_DL_002620, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { object_haka_door_DL_003890, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { object_ice_objects_DL_001D10, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { gGTGDoorDL, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { gGanonsCastleDoorDL, gDoorMetalBarsDL, 130, 12, 20, 15 },
-    { object_ouke_haka_DL_0000C0, gDoorMetalBarsDL, 130, 12, 20, 15 },
+    /* 0x0000 */ Gfx* doorDL;
+    /* 0x0004 */ Gfx* barsDL;
+    /* 0x0008 */ u8 barsOpenOffsetY;
+    /* 0x0009 */ u8 barsOffsetZ;
+    /* 0x000A */ u8 rangeSides;
+    /* 0x000B */ u8 rangeY;
+} DoorShutterGfxInfo;
+
+static DoorShutterGfxInfo sGfxInfo[] = {
+    { gDTDungeonDoor1DL, gDoorMetalBarsDL, 130, 12, 20, 15 },                      // DOORSHUTTER_GFX_DEKU_TREE_0
+    { gDTDungeonDoor2DL, gDoorMetalBarsDL, 130, 12, 20, 15 },                      // DOORSHUTTER_GFX_DEKU_TREE_1
+    { gDodongoDoorDL, gDodongoBarsDL, 240, 14, 70, 15 },                           // DOORSHUTTER_GFX_DODONGOS_CAVERN
+    { gJabuDoorSection1DL, gJabuWebDoorDL, 0, 110, 50, 15 },                       // DOORSHUTTER_GFX_JABU_JABU
+    { gPhantomGanonBarsDL, NULL, 130, 12, 50, 15 },                                // DOORSHUTTER_GFX_PHANTOM_GANON_BARS
+    { gGohmaDoorDL, NULL, 130, 12, 50, 15 },                                       // DOORSHUTTER_GFX_GOHMA_BLOCK
+    { gSpiritDoorDL, gJyaDoorMetalBarsDL, 240, 14, 50, 15 },                       // DOORSHUTTER_GFX_SPIRIT_TEMPLE
+    { gBossDoorDL, NULL, 130, 12, 50, 15 },                                        // DOORSHUTTER_GFX_BOSS_DOOR
+    { gDungeonDoorDL, gDoorMetalBarsDL, 130, 12, 20, 15 },                         // DOORSHUTTER_GFX_8
+    { gFireTempleDoorFrontDL, gDoorMetalBarsDL, 130, 12, 20, 15 },                 // DOORSHUTTER_GFX_FIRE_TEMPLE_9
+    { gFireTempleDoorBackDL, gDoorMetalBarsDL, 130, 12, 20, 15 },                  // DOORSHUTTER_GFX_FIRE_TEMPLE_10
```

Comment:
Very minor but maybe label these `_1` and `_2` for each scene?
(Though as far as I can tell these are all "front" and "back" styles)